sdk/src/azure/platform/az_posix.c

```diff
@@ -1,21 +1,25 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
-#include <azure/core/internal/az_config_internal.h>
 #include <azure/core/az_platform.h>
+#include <azure/core/internal/az_config_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
 
 #include <time.h>
 
 #include <unistd.h>
 
 #include <azure/core/_az_cfg.h>
 
-AZ_NODISCARD int64_t az_platform_clock_msec()
+AZ_NODISCARD az_result az_platform_clock_msec(int64_t* out_clock_msec)
 {
-  return (int64_t)((clock() / CLOCKS_PER_SEC) * _az_TIME_MILLISECONDS_PER_SECOND);
+  _az_PRECONDITION_NOT_NULL(out_clock_msec);
+  *out_clock_msec = (int64_t)((clock() / CLOCKS_PER_SEC) * _az_TIME_MILLISECONDS_PER_SECOND);
+  return AZ_OK;
 }
 
-void az_platform_sleep_msec(int32_t milliseconds)
+AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
 {
   (void)usleep((useconds_t)milliseconds * _az_TIME_MICROSECONDS_PER_MILLISECOND);
```

Comment:
unrelated: precondition that `milliseconds` is positive.

Reply:
So far we say in the docs, `@remarks The behavior is undefined when \p milliseconds is a non-positive value (0 or less than 0)`. That's because on some platform, it would yield to other thread, but will return control back if no other thread needs it. We don't use this feature though.

Reply:
> That's because on some platform, it would yield to other thread, but will return control back if no other thread needs it. We don't use this feature though.

We can add a precondition now to help folks right correct code, and alter loosen that in the future, if that value becomes meaningful.

Reply:
> > That's because on some platform, it would yield to other thread, but will return control back if no other thread needs it. We don't use this feature though.
> 
> We can add a precondition now to help folks write correct code, and loosen that in the future, if that value becomes meaningful.

We could precondition, but we would be limiting to a particular implementation of POSIX.  Its not clear if every implementation places the same constraints.  e.g. some platforms say the `usec is not smaller than 1000000` others allow 0 to mean yield.

Reply:
This is **one particular** implementation though. Adding a precondition here doesn't impact users who want to provide their own PAL implementation. Passing in a negative millisecond will overflow when cast to unsigned int, and that could be an unintended bug in the code (given `usleep` takes an unsigned number).